src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -96,6 +96,12 @@ trait GenTraversableOnce[+A] extends Any {
    */
   def size: Int
 
+   /** The size of this $coll if it is can be cheaply computed
+   *
+   *  @return    the number of elements in this $coll, or -1 if the size cannot be determined cheaply
+   */
+  protected[collection] def sizeHint: Int = -1
```

Comment:
I'm not sure about restricting the visibility of this method. Supposedly, if I create my own type of collection, I might want to call `sizeHint` on another collection? Or should I only ever call the methods `sizeHint(coll)` & co.?